logic/src/commonMain/kotlin/com/wire/kalium/logic/CoreLogic.kt

```diff
@@ -11,7 +12,8 @@ import com.wire.kalium.network.LoginNetworkContainer
 class CoreLogic(
     private val clientLabel: String,
     private val rootProteusDirectoryPath: String,
-    private val authSessionMapper: AuthSessionMapper = AuthSessionMapper()
+    private val authSessionMapper: AuthSessionMapper = AuthSessionMapper(),
+    private val clientConfig: ClientConfig
```

Comment:
Can we leave the optional parameter to the end? Just to avoid naming parameters unnecessarily.

Reply:
I am thinking about a different approach, like making `CoreLogic` platform dependent and the android class will require `Context` since context is needed in many different other classes E.g. DataStore

Reply:
That can also work, but maybe it grows this PR too big? I'd suggest changing these platform-specific things in another PR.

In any case, I would not expect anyone using `CoreLogic` to actually create a `AuthSessionMapper`. Especially when it's a closed class.

Currently something like this is needed:
```kotlin
val core = CoreLogic("Kalium Sample App", context.file("proteusRoot"), clientConfig = ClientConfig())
```

The `clientConfig = ` is something I'm trying to avoid, because `AuthSessionMapper` gets in the way. We don't provide good discoverability to whoever would be using this code.